sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -4007,6 +4009,16 @@ object ApplyCharTypePadding extends Rule[LogicalPlan] {
             b.withNewChildren(newChildren.reverse)
           }.getOrElse(b)
 
+        case b @ BinaryComparison(OuterReference(attr: Attribute), lit) if lit.foldable =>
+          padOuterAttrLitComp(attr, lit).map { newChildren =>
```

Comment:
shall we let `padAttrLitCmp` to take `Expression` and `Metadata`, instead of `Attribute`, so that we can pass `OuterReference` to `padAttrLitCmp` directly and we don't need to create `padOuterAttrLitComp`?

Reply:
done